src/pages/Detail/DetailTop.js

```diff
@@ -1,54 +1,121 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 
+import AmountOption from './AmountOption';
+
+// select -> state
+// 함수를 만들어서 자식한테 전달
+// 자식에서 함수기 실행되면은 select된 값을 변경
+
 export class DetailTop extends Component {
+  //state초기화
+  constructor(props) {
+    super(props);
+    this.state = {
+      numValue: 1,
+      isOn: false,
+      amountOption: [],
+      selectedOption: this.props.productOption[0],
+      productOption: '',
+      option: '',
+    };
+    this.onClickToggleHandler = this.onClickToggleHandler.bind(this);
+  }
+  //수량 증가 감소
+  onClickHandler = () => {
+    this.setState({ numValue: this.state.numValue + 1 });
+  };
+  onClickhandlerRemove = () => {
+    //const { numValue } = this.state.numValue;
+    if (this.state.numValue > 1) {
+      this.setState({ numValue: this.state.numValue - 1 });
+    }
+  };
+  //종류 추가
+  onClickToggleHandler = () => {
+    //const { isOn } = this.state.isOn;
+    //console.log(isOn);
+    this.setState({
+      isOn: !this.state.isOn,
+    });
+  };
+
+  selectOption = option => {
+    this.setState({
+      selectedOption: option,
+    });
+    this.onClickToggleHandler();
+  };
+
   render() {
+    const productInfo = this.props;
+    const selectedOption = this.state.selectedOption;
+    //console.log(productInfo.productOption);
+    //console.log(this.state.productOption);
+    console.log(selectedOption, 'dfdfd');
     return (
-      <section className="detail-top-wrap">
+      <section className="detail-top-wrap" key="{productInfo.key}">
         <div className="detail-top-container">
           <div className="detail-top-data">
             <picture>
-              <img
-                src="https://firebasestorage.googleapis.com/v0/b/jyg-custom-seoul-app/o/frontend%2Fthumbnails%2Ftransparent_background%2Fmanilaclam-water-detail.png?alt=media"
-                alt="제품 이미지"
-              />
+              <img src={productInfo.productImg} alt="제품 이미지" />
```

Comment:
- 위에서 props를 구조 분해 할당 해주시면 다음과 같이 작성 하실 수 있습니다!
```suggestion
              <img src={productImg} alt="제품 이미지" />
```

Reply:
해당부분도 this.props로 풀어서 썻습니다. 다시 한번 확인부탁드리겠습니다.